workers/unity/Assets/Gdk/Core/GameObjectRepresentation/SpatialOSGameObjectCreator.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Creates a companion gameobject for newly spawned entities according to a prefab definition.
+    /// </summary>
+    public class SpatialOSGameObjectCreator
+    {
+        private readonly World world;
+        private readonly Dictionary<string, GameObject> cachedPrefabs;
+
+        public SpatialOSGameObjectCreator(World world, Dictionary<string, GameObject> initialCachedPrefabs)
+        {
+            this.world = world;
+            cachedPrefabs = initialCachedPrefabs;
+        }
+
+        public GameObject CreateSpatialOSGameObject(Entity entity, string prefabName, Vector3 position, Quaternion rotation, ViewCommandBuffer viewCommandBuffer, long spatialEntityId)
+        {
+            GameObject prefab;
+            if (!cachedPrefabs.TryGetValue(prefabName, out prefab))
```

Comment:
Is this `prefabName` a name or a path? Need to make consistent with the error message either way.

Reply:
The error message refers to a `prefabPath` for example

Reply:
Good point. Renamed to `prefabPath`.